File: Dal/Models/UserSlot.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using BPGezinswetenschappen.DAL.Models;

namespace DAL.Models
{
    public class UserSlot
    {
        public int UserId { get; set; }
        public User User { get; set; } = null!;

        public int SlotId { get; set; }
        public Slot Slot { get; set; } = null!;

        public int? RoleId { get; set; }
        public Role? Role { get; set; }

    }
}

File: Client/src/app/shared/models/userslot.ts
import { Slot } from './slot';
import { User } from './user';

// Placeholder for UserSlot model. Update fields as needed.
export interface UserSlot {
  userId: number;
  user: User; // Use 'any' to avoid circular reference, or import User if needed
  slotId: number;
  slot: Slot; // Use 'any' to avoid circular reference, or import Slot if needed
  roleId?: number;
  role?: any; // Use 'any' to avoid circular reference, or import Role if needed
}
